### PR TITLE
Fix dataset matcher when no tool available.

### PR DESCRIPTION
**Pull request by zipho**
Broken with 80ac816af94d43d7214671582db127483baea2d3.

### CODE REVIEW

**Comment by jmchilton** [comment]
@galaxybot test this

**Comment by martenson** [comment]
@galaxybot test this

**Comment by guerler** [comment]
Fix works for me. Thanks.

File: lib/galaxy/tools/parameters/dataset_matcher.py [hunk 1]
```diff
@@ -35,7 +35,12 @@ def hda_accessible( self, hda, check_security=True ):
         accessible to user.
         """
         dataset = hda.dataset
-        state_valid = dataset.state in self.tool.valid_input_states
+        has_tool = self.tool
+        if has_tool:
```
Comment (nsoranzo) [inline lib/galaxy/tools/parameters/dataset_matcher.py:39]: Out of curiosity, when did you see this to be `None`?

Comment (guerler) [inline lib/galaxy/tools/parameters/dataset_matcher.py:39]: I noticed that loading workflows for execution fails without this fix, possibly due workflow input modules i.e. those have no associated tools, but I did not look further into it.

Comment (nsoranzo) [inline lib/galaxy/tools/parameters/dataset_matcher.py:39]: Right, found an example at https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/workflow/modules.py#L532 , thanks @guerler!

**Pull request merged**